sympy/physics/optics/medium.py

```diff
@@ -69,23 +70,32 @@ class Medium(Symbol):
     """
 
     def __new__(cls, name, permittivity=None, permeability=None, n=None):
-        obj = super().__new__(cls, name)
-        obj._permittivity = sympify(permittivity)
-        obj._permeability = sympify(permeability)
-        obj._n = sympify(n)
+
+        if not isinstance(name, Str):
+            name = Str(name)
+
         if n is not None:
             if permittivity is not None and permeability is None:
-                obj._permeability = n**2/(c**2*obj._permittivity)
+                permeability = n**2/(c**2*permittivity)
             if permeability is not None and permittivity is None:
-                obj._permittivity = n**2/(c**2*obj._permeability)
+                permittivity = n**2/(c**2*permeability)
             if permittivity is not None and permittivity is not None:
-                if abs(n - c*sqrt(obj._permittivity*obj._permeability)) > 1e-6:
-                   raise ValueError("Values are not consistent.")
+                expr = abs(n - c*sqrt(permittivity*permeability))
+                expr = expr.subs({meter: 1, second: 1})
+                if len(expr.free_symbols) == 0 and expr > 1e-6:
+                    raise ValueError("Values are not consistent.")
         elif permittivity is not None and permeability is not None:
-            obj._n = c*sqrt(permittivity*permeability)
+            n = c*sqrt(permittivity*permeability)
         elif permittivity is None and permeability is None:
-            obj._permittivity = _e0mksa
-            obj._permeability = _u0mksa
+            permittivity = _e0mksa
+            permeability = _u0mksa
+            n = c*sqrt(permittivity*permeability)
+        args = list(map(sympify, (permittivity, permeability, n)))
```

Comment:
`_sympify` is prefered over `sympify` (it is stricter and shouldn't return non-Basic types, which `sympify` might)

Reply:
Alright, I'll make the change.